source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/utils/hydra.py

```diff
@@ -0,0 +1,98 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""Sub-module with utilities for the hydra configuration system."""
+
+
+import functools
+from collections.abc import Callable
+
+try:
+    import hydra
+    from hydra.core.config_store import ConfigStore
+    from omegaconf import DictConfig, OmegaConf
+except ImportError:
+    raise ImportError("Hydra is not installed. Please install it by running 'pip install hydra-core'.")
+
+from omni.isaac.lab.envs import DirectRLEnvCfg, ManagerBasedRLEnvCfg
+from omni.isaac.lab.utils import replace_slices_with_strings, replace_strings_with_slices
+
+from omni.isaac.lab_tasks.utils.parse_cfg import load_cfg_from_registry
+
+
+def register_task_to_hydra(
+    task_name: str, agent_cfg_entry_point: str
+) -> tuple[ManagerBasedRLEnvCfg | DirectRLEnvCfg, dict]:
+    """Register the task configuration to the Hydra configuration store.
+
+    This function resolves the configuration file for the environment and agent based on the task's name.
+    It then registers the configurations to the Hydra configuration store.
+
+    Args:
+        task_name: The name of the task.
+        agent_cfg_entry_point: The entry point key to resolve the agent's configuration file.
+
+    Returns:
+        A tuple containing the parsed environment and agent configuration objects.
+    """
+    # load the configurations
+    env_cfg = load_cfg_from_registry(task_name, "env_cfg_entry_point")
+    agent_cfg = load_cfg_from_registry(task_name, agent_cfg_entry_point)
+    # convert the configs to dictionary
+    env_cfg_dict = env_cfg.to_dict()
+    if isinstance(agent_cfg, dict):
+        agent_cfg_dict = agent_cfg
+    else:
+        agent_cfg_dict = agent_cfg.to_dict()
+    cfg_dict = {"env": env_cfg_dict, "agent": agent_cfg_dict}
+    # replace slices with strings because OmegaConf does not support slices
+    cfg_dict = replace_slices_with_strings(cfg_dict)
+    # store the configuration to Hydra
+    ConfigStore.instance().store(name=task_name, node=cfg_dict)
+    return env_cfg, agent_cfg
+
+
+def hydra_task_config(task_name: str, agent_cfg_entry_point: str) -> Callable:
+    """Decorator to handle the Hydra configuration for a task.
+
+    This decorator registers the task to Hydra and updates the environment and agent configurations from Hydra parsed
+    command line arguments.
+
+    Args:
+        task_name: The name of the task.
+        agent_cfg_entry_point: The entry point key to resolve the agent's configuration file.
+
+    Returns:
+        The decorated function with the envrionment's and agent's configurations updated from command line arguments.
+    """
+
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            # register the task to Hydra
+            env_cfg, agent_cfg = register_task_to_hydra(task_name, agent_cfg_entry_point)
+
+            # define thr new Hydra main function
+            @hydra.main(config_path=None, config_name=task_name, version_base="1.3")
```

Comment:
Here, it is also possible to use the compose API: https://hydra.cc/docs/advanced/compose_api/

The minimal example originally done for ray: https://github.com/facebookresearch/hydra/blob/main/examples/advanced/ray_example/ray_compose_example.py